cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -367,6 +368,10 @@ private void restartIfNecessary(NodeRef nodeRef, RestartContext restartContext)
             throw new UnforceableProblem("Error getting pod " + nodeRef.podName(), e);
         }
 
+        if (kafkaAgentClient == null) {
+            this.kafkaAgentClient = initKafkaAgentClient();
+        }
```

Comment:
Same as in the other class. Why not do this in the constructor?

Reply:
Similar reason above applies to this. I wasn't sure if when we can't initialise kafkaAgentClient, that should cause the reconciliation to start. Admin clients are not initialised from the constructor, so kafkaAgentClient probably should not either.

Reply:
I can add comments to clarify this if you agree

Reply:
Ok, thanks.